oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/spanattach/SpanAttachedEventRecord.java

```diff
@@ -0,0 +1,112 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.analysis.manual.spanattach;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.skywalking.oap.server.core.Const;
+import org.apache.skywalking.oap.server.core.analysis.Stream;
+import org.apache.skywalking.oap.server.core.analysis.record.Record;
+import org.apache.skywalking.oap.server.core.analysis.worker.RecordStreamProcessor;
+import org.apache.skywalking.oap.server.core.source.ScopeDeclaration;
+import org.apache.skywalking.oap.server.core.storage.annotation.BanyanDB;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.core.storage.type.Convert2Entity;
+import org.apache.skywalking.oap.server.core.storage.type.Convert2Storage;
+import org.apache.skywalking.oap.server.core.storage.type.StorageBuilder;
+
+import static org.apache.skywalking.oap.server.core.source.DefaultScopeDefine.SPAN_ATTACHED_EVENT;
+
+@Setter
+@Getter
+@ScopeDeclaration(id = SPAN_ATTACHED_EVENT, name = "SpanAttachedEvent")
+@Stream(name = SpanAttachedEventRecord.INDEX_NAME, scopeId = SPAN_ATTACHED_EVENT, builder = SpanAttachedEventRecord.Builder.class, processor = RecordStreamProcessor.class)
+public class SpanAttachedEventRecord extends Record {
+
+    public static final String INDEX_NAME = "span_attached_event_record";
+    public static final String START_TIME_SECOND = "start_time_second";
+    public static final String START_TIME_NANOS = "start_time_nanos";
+    public static final String EVENT = "event";
+    public static final String END_TIME_SECOND = "end_time_second";
+    public static final String END_TIME_NANOS = "end_time_nanos";
+    public static final String TRACE_REF_TYPE = "trace_ref_type";
+    public static final String TRACE_ID = "trace_id";
+    public static final String TRACE_SEGMENT_ID = "trace_segment_id";
+    public static final String TRACE_SPAN_ID = "trace_span_id";
+    public static final String DATA_BINARY = "data_binary";
+
+    @Column(columnName = START_TIME_SECOND)
+    private long startTimeSecond;
+    @Column(columnName = START_TIME_NANOS)
+    private int startTimeNanos;
+    @Column(columnName = EVENT)
+    private String event;
+    @Column(columnName = END_TIME_SECOND)
+    private long endTimeSecond;
+    @Column(columnName = END_TIME_NANOS)
+    private int endTimeNanos;
+    @Column(columnName = TRACE_REF_TYPE)
+    private int traceRefType;
```

Comment:
Should we add a `SpanReferenceType` type field(not a column` in the Record?

Reply:
Updated it as an enum.